### PR TITLE
fix: move signaturetest to internal package

### LINKED ISSUE #104: signaturetest package should not be exported
Currently, the signature/signaturetest is an exported package. It should be moved to an internal folder to avoid using by public users.

### PR DESCRIPTION
**Pull request by JeyJeyGao**
the signaturetest package only used by signature unit test, so it should be moved to internal package.
Also added unit test to increase the code coverage.

Resolves #104 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: signature/internal/signaturetest/algorithm.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// Package signaturetest includes variables and functions for signature
```
Comment (shizhMSFT) [inline signature/internal/signaturetest/algorithm.go:1]: If it is for internal test purposes only, is it possible to name `algorithm.go` to `algorithm_test.go`, `signer.go` to `signer_test.go`?

Comment (JeyJeyGao) [reply signature/internal/signaturetest/algorithm.go:1]: I tried it. However, it looks like variables and functions in a file endded with "_test.go" cannot be used out side the packge. Those functions are shared by signature/jws and signature/cose packages.

Comment (shizhMSFT) [reply signature/internal/signaturetest/algorithm.go:1]: I see. Can we also write unit tests for those "test functions"? It will also impact the coverage.

Comment (JeyJeyGao) [reply signature/internal/signaturetest/algorithm.go:1]: Added test cases.

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**